File: advent6a.py
def solution(themap: list):
    n = len(themap)
    m = len(themap[0])
    visited = [[False for _ in range(m)] for _ in range(n)]
    start = [0, 0]
    ans = 0

    for i in range(n):
        for j in range(m):
            if themap[i][j] in ("^", ">", "v", "<"):
                start = [i, j]

    while True:
        i, j = start
        if not visited[i][j]:
            ans += 1
        visited[i][j] = True

        if themap[i][j] == "^":
            if i - 1 < 0:
                return ans
            if themap[i - 1][j] == "#":
                themap[i][j] = ">"
            else:
                themap[i][j] = '.'
                themap[i-1][j] = '^'
                start = [i - 1, j]
        elif themap[i][j] == ">":
            if j + 1 >= m:
                return ans
            if themap[i][j + 1] == "#":
                themap[i][j] = "v"
            else:
                themap[i][j] = '.'
                themap[i][j+1] = '>'
                start = [i, j + 1]
        elif themap[i][j] == "v":
            if i + 1 >= n:
                return ans
            if themap[i + 1][j] == "#":
                themap[i][j] = "<"
            else:
                themap[i][j] = '.'
                themap[i+1][j] = 'v'
                start = [i + 1, j]
        elif themap[i][j] == "<":
            if j - 1 < 0:
                return ans
            if themap[i][j - 1] == "#":
                themap[i][j] = "^"
            else:
                themap[i][j] = '.'
                themap[i][j-1] = '<'
                start = [i, j - 1]


with open("input.txt", "r") as fp:
    print(solution([[c for c in line.strip("\n")] for line in fp.readlines()]))

File: aoc2025/6a/advent6a.java
import java.io.File;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Scanner;

public class advent6a {
    private static List<List<Integer>> groups = new ArrayList<>();
    private static List<Integer> operators = new ArrayList<>();

    private static long solution() {
        long ans = 0L;

        for (int i = 0; i < groups.size(); i++) {
            long val = 0;
            if (operators.get(i) == 0)
                val = 1;

            for (int x : groups.get(i)) {
                if (operators.get(i) == 0) {
                    val *= x;
                } else {
                    val += x;
                }
            }
            ans += val;
        }

        return ans;
    }

    public static void main(String[] args) {
        try {
            File fp = new File("input.txt");
            Scanner fpr = new Scanner(fp);

            while (fpr.hasNext()) {
                String line = fpr.nextLine();
                if (!fpr.hasNext()) {
                    for (String operator : line.split(" ")) {
                        if (operator.isEmpty())
                            continue;
                        operators.add("*+".indexOf(operator));
                    }
                    break;
                }

                List<Integer> nums = Arrays.stream(line.split(" ")).filter(x -> !x.isEmpty()).map(Integer::parseInt)
                        .toList();

                if (groups.size() == 0) {
                    for (int i = 0; i < nums.size(); i++) {
                        groups.add(new ArrayList<>());
                    }
                }

                for (int i = 0; i < nums.size(); i++) {
                    groups.get(i).add(nums.get(i));
                }
            }

            fpr.close();

            System.out.println(solution());
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}
